add-on/TA-Demisto/bin/demisto_setup.py

```diff
@@ -91,80 +89,90 @@ def handleEdit(self, confInfo):
                 url += ":" + self.callerArgs.data['PORT'][0]
 
             '''
-                Create Test Incident to demisto to verify if the configuration entered are correct
+                Create Test Incident to demisto to verify if the configuration is correct
                 Store configuration only if create incident was successful.
             '''
             url += "/incidenttype"
```

Comment:
Instead of directly going to `/incidenttype` that have authorization and can fail for multiple reasons, I suggest we will make the following calls so we will give a proper message to the user what have failed.
1. `/proxyMode` - no auth, so if we fail here it is just about network / certificate.
2. `/user` - will let us check auth and that this user have the required permissions.

For each case we should send a message to the user with the exact reason, ad if we have a suggestion include it.

Reply:
@orius123 added.